packages/amplify-category-geo/src/provider-utils/awscloudformation/index.ts

```diff
@@ -0,0 +1,228 @@
+ */
+export async function addResource(
+  context: $TSContext,
+  service: string,
+  parameters?: Partial<MapParameters>
+): Promise<string> {
+  // load the service config for this service
+  const BAD_SERVICE_ERR = new Error(`amplify-category-geo is not configured to provide service type ${service}`);
+
+  while (!checkIfAuthExists(context)) {
+    if (
+      await context.amplify.confirmPrompt(
+        'You need to add auth (Amazon Cognito) to your project in order to add geo resources. Do you want to add auth now?',
+      )
+    ) {
+      await context.amplify.invokePluginMethod(context, 'auth', undefined, 'add', [context]);
+      break;
+    } else {
+      context.usageData.emitSuccess();
+      exitOnNextTick(0);
+    }
+  }
+
+  switch (service) {
+    case ServiceName.Map:
+      const serviceConfig: ServiceConfig<MapParameters> = supportedServices[service];
+      return addMapResource(context, service, serviceConfig, parameters);
+    default:
+      throw BAD_SERVICE_ERR;
+  }
+}
+
+/**
+ * Entry point for updating existing Geo resource
+ * @param context Amplify Core Context object
+ * @param service The cloud service that is providing the category
+ * @param parameters Parameters used to configure the resource. If not specified, a walkthrough will be launched to populate it.
+ */
+export async function updateResource(
+  context: any,
+  service: string
+): Promise<string> {
+  // load the service config for this service
+  const BAD_SERVICE_ERR = new Error(`amplify-category-geo is not configured to provide service type ${service}`);
+
+  switch (service) {
+    case ServiceName.Map:
+      const serviceConfig: ServiceConfig<MapParameters> = supportedServices[service];
+      return updateMapResource(context, service, serviceConfig);
+    default:
+      throw BAD_SERVICE_ERR;
+  }
+};
+
+export async function removeResource(
+  context: any,
+  service: string
+): Promise<string> {
+  // load the service config for this service
+  const BAD_SERVICE_ERR = new Error(`amplify-category-geo is not configured to provide service type ${service}`);
+
+  switch (service) {
+    case ServiceName.Map:
+      return removeMapResource(context, service);
+    default:
+      throw BAD_SERVICE_ERR;
+  }
+};
+
+function checkIfAuthExists(context: $TSContext) {
+    const { amplify } = context;
+    const { amplifyMeta } = amplify.getProjectDetails();
+    let authExists = false;
+    const authServiceName = 'Cognito';
+    const authCategory = 'auth';
+
+    if (amplifyMeta[authCategory] && Object.keys(amplifyMeta[authCategory]).length > 0) {
+      const categoryResources = amplifyMeta[authCategory];
+      Object.keys(categoryResources).forEach(resource => {
+        if (categoryResources[resource].service === authServiceName) {
+          authExists = true;
+        }
+      });
+    }
+    return authExists;
+}
+
+export async function addMapResource(
+  context: $TSContext,
+  service: string,
```

Comment:
type should be `ServiceName`

Also, it looks like this will always be `ServiceName.Map` in which case it shouldn't be a parameter at all

Reply:
removed the param as it was redundant